File: coffeetime-parent/RubyGun/src/24_date.rb
require 'Time'

# Current Time
time1 = Time.new
puts time1.inspect

time2 = Time.now
puts time2.inspect

# Timestamp
puts time1.to_i

# Query Fields
puts time1.year
puts time1.month
puts time1.day
puts time1.wday

# Time Zone
puts Time.local(2008, 7, 8)
puts Time.local(2008, 7, 8, 9, 10)
puts Time.utc(2008, 7, 8, 9, 10)

# Format
puts time1.strftime('%Y-%m-%d %H:%M:%S')
puts Time.parse('2014-04-23').inspect

# Calc
puts time2 - 10 # 10 seconds ago

File: coffeetime-parent/Python3Gun/src/24_date.py
import datetime
import time as timer

# Current Time
print(timer.ctime())
print(datetime.datetime.now())

# Timestamp
print(timer.time())

# Query Fields
t = datetime.datetime.now()
print(t.year)
print(t.month)
print(t.day)

# Format
print(timer.strftime('%Y-%m-%d %H:%M:%S'))
print(timer.strptime('2014-04-23', '%Y-%m-%d'))

# Calc
print(t - datetime.timedelta(seconds=-10))
